From: Jesse Hallett <jesse@sitr.us>
Date: Sat, 15 Dec 2018 15:27:34 -0500
Subject: [PATCH] Add type annotation to Unstated for Flow versions >=0.85.0

This patch will be applied to the Unstated dependency automatically whenever
packages are updated thanks to the `"prepare": "patch-package"` script in
`package.json`.

Starting in version 0.85.0 Flow requires type annotations in positions that it
not require previously. This patch may be deleted after upgrading to a version
of Unstated that includes this pull request:

https://github.com/jamiebuilds/unstated/pull/69

For background on the change to Flow see:

https://medium.com/flow-type/asking-for-required-annotations-64d4f9c1edf8

patch-package
--- a/node_modules/unstated/lib/unstated.js.flow
+++ b/node_modules/unstated/lib/unstated.js.flow
@@ -1,10 +1,10 @@
 // @flow
 import React, { type Node } from 'react';
-import createReactContext from 'create-react-context';
+import createReactContext, { type Context } from 'create-react-context';
 
 type Listener = () => mixed;
 
-const StateContext = createReactContext(null);
+const StateContext: Context<ContainerMapType | null> = createReactContext(null);
 
 export class Container<State: {}> {
   state: State;
